core/domain/email_manager.py

```diff
@@ -164,5 +178,71 @@ def send_post_signup_email(user_id):
         EMAIL_FOOTER.value)
 
     _send_email(
-        user_id, feconf.SYSTEM_COMMITTER_ID, email_models.INTENT_SIGNUP,
+        user_id, feconf.SYSTEM_COMMITTER_ID, feconf.EMAIL_INTENT_SIGNUP,
         email_subject, email_body)
+
+
+def require_valid_intent(intent):
+    if intent not in feconf.VALID_MODERATOR_ACTIONS:
+        raise Exception('Unrecognized email intent: %s' % intent)
+
+
+def _get_email_config(intent):
```

Comment:
It actually took me a while to hit on this idea, because somewhere along the line I needed to change 'user action' into 'intent of email being sent' -- and this is what this represents. I think this is a good solution, and that the intent/action string being the same for the existing actions is only coincidental. They really represent different things conceptually, so I think that having this seam is fine.